aws/resource_aws_flow_log.go

```diff
@@ -18,45 +19,71 @@ func resourceAwsFlowLog() *schema.Resource {
 		Importer: &schema.ResourceImporter{
 			State: schema.ImportStatePassthrough,
 		},
+		MigrateState:  resourceAwsFlowLogMigrateState,
+		SchemaVersion: 1,
 
 		Schema: map[string]*schema.Schema{
+
 			"iam_role_arn": &schema.Schema{
 				Type:     schema.TypeString,
-				Required: true,
+				Optional: true,
 				ForceNew: true,
 			},
 
-			"log_group_name": &schema.Schema{
+			"log_destination": &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validateArn,
+			},
+
+			"log_destination_type": &schema.Schema{
 				Type:     schema.TypeString,
-				Required: true,
+				Default:  ec2.LogDestinationTypeCloudWatchLogs,
+				Optional: true,
 				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					ec2.LogDestinationTypeCloudWatchLogs,
+					ec2.LogDestinationTypeS3,
+				}, false),
 			},
 
-			"vpc_id": &schema.Schema{
-				Type:          schema.TypeString,
-				Optional:      true,
-				ForceNew:      true,
-				ConflictsWith: []string{"subnet_id", "eni_id"},
+			"log_group_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
 			},
 
-			"subnet_id": &schema.Schema{
-				Type:          schema.TypeString,
-				Optional:      true,
-				ForceNew:      true,
-				ConflictsWith: []string{"eni_id", "vpc_id"},
+			"resource_id": &schema.Schema{
```

Comment:
Should be named `resource_ids` as potentially multiple IDs can be specified?

Reply:
I did this to conform to the prior code but will make the change